src/transformers/modeling_t5.py

```diff
@@ -421,31 +478,63 @@ def forward(
         encoder_attention_mask=None,
         encoder_decoder_position_bias=None,
         head_mask=None,
+        past_key_value_state=None,
     ):
+
+        if self.is_decoder and self.output_past and past_key_value_state is not None:
+            assert (
+                len(past_key_value_state) == 4
+            ), "The should be 4 past states. 2 (past / key) for self attention. 2 (past / key) for cross attention. Got {} past key / value states".format(
+                len(past_key_value_state)
+            )
```

Comment:
We store both the key value vectors of both the self attn layer and the cross attn layer so that we have 4 stored states per layer.